cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java

```diff
@@ -430,6 +433,47 @@ public void testPullAppPullFailureException() throws Exception {
     Assert.assertTrue(response.getResponseBodyAsString().contains("Failed to pull application"));
   }
 
+  @Test
+  public void testListAppsSucceed() throws Exception {
+    ListAppResponse app1 = new ListAppResponse("app1", "hash1");
+    ListAppResponse app2 = new ListAppResponse("app2", "hash2");
+    ListAppsResponse expectedListResult = new ListAppsResponse(Arrays.asList(app1, app2));
+    Mockito.doReturn(expectedListResult).when(sourceControlService)
+      .listApps(Mockito.any());
+
+    HttpResponse response = listApplicationsFromRepository(Id.Namespace.DEFAULT.getId());
+    assertResponseCode(200, response);
+    ListAppsResponse result = readResponse(response, ListAppsResponse.class);
+
+    Assert.assertEquals(result.getApps(), expectedListResult.getApps());
+  }
+
+  @Test
+  public void testListAppsFailed() throws Exception {
+    Mockito.doThrow(ListAppsFailureException.class).when(sourceControlService)
+      .listApps(Mockito.any());
+
+    HttpResponse response = listApplicationsFromRepository(Id.Namespace.DEFAULT.getId());
+    assertResponseCode(500, response);
+  }
+
+  @Test
+  public void testListAppsFailedAuth() throws Exception {
+    Mockito.doThrow(AuthenticationConfigException.class).when(sourceControlService)
+      .listApps(Mockito.any());
+
+    HttpResponse response = listApplicationsFromRepository(Id.Namespace.DEFAULT.getId());
+    assertResponseCode(400, response);
```

Comment:
Why are we sending 400 here? The request itself is fine, it's server misconfiguration issue

Reply:
We are currently sending back 400 when handler gets `AuthenticationConfigException` due to missing securestore entry for token. As this is a user provided configuration we are treating this as `400 Bad Request`

Reply:
After the change, it should assert 500 I believe?